// components/Button.js
import React from 'react'
import { useLocation } from 'react-router'

import { Icon } from 'components/Icon'

export const Button = ({
    kind,
    text,
    leftIcon,
    rightIcon,
    round,
    ...rest
}) => {
    const renderLeftIcon = leftIcon ? (
        <Icon
            name={ leftIcon }
            size='30'
        ></Icon>
    ) : null
    const renderRightIcon = rightIcon ? (
        <Icon
            name={ rightIcon }
            size='30'
        ></Icon>
    ) : null

    return (
        <button
            className={ 'button' + ( kind ? ` ${kind}` : ' secondary')}
            style={ round ? {borderRadius: '6px'} : null}
            {...rest}
        >
            { renderLeftIcon }
            <span>{text}</span>
            { renderRightIcon }
        </button>
    );
}

export const WishButton = ({ wish, kind }) => {
    const loca = useLocation().pathname;
    const isMyWish = loca.split('/').at(1) === 'my-wishes';

    const content = (wish.state === 'actual' && isMyWish)
        ? {
            icon: 'ok',
            text: 'Отметить исполненным',
            handleClick: (e) => { e.stopPropagation() },
        }
        : (wish.state === 'actual' && !isMyWish)
        ? {
            icon: 'present',
            text: 'Исполнить желание',
        }
        : (wish.state === 'reserved')
        ? {
            icon: 'lock',
            text: 'Зарезервировано',                
            disabled: true,
        }
        : (wish.state === 'completed')
        ? {
            icon: 'ok',
            text: 'Желание исполнено',
            disabled: true,
        }
        : undefined

    if(!content) return null;

    return (
        <Button
            kind={ kind || 'secondary' }
            leftIcon={ content.icon }
            text={ content.text }
            disabled={ !!content.disabled }
            onClick={ content.handleClick }
        />
    );
}
