// components/Highlighter.js
import React from 'react'
import { useSelector } from 'react-redux'

export const Goo = () => (
    <svg xmlns='http://www.w3.org/2000/svg' version='1.1' style={{ width: 0, height: 0}}>
        <defs>
            <filter id='goo'>
                <feGaussianBlur in='SourceGraphic' stdDeviation='10' result='blur'/>
                <feColorMatrix in='blur' type='matrix' values='1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 18 -7' result='goo' />
                <feBlend in='SourceGraphic' in2='goo'/>
            </filter>
        </defs>
    </svg>
);

export const Highlighter = ({ location, className }) => {
    const leftSectionSlider = useSelector(state => state.sliders.leftSectionSlider);
    const rightSectionSlider = useSelector(state => state.sliders.rightSectionSlider);
    const modeToggleCoords = useSelector(state => state.highlighters.inModeToggleCoords);
    const navBarCoords = useSelector(state => state.highlighters.inNavBarCoords);
    const defaultCoords = {x:0, y:0, width:0, height:0};

    const coords = (location==='left-section-slider')
                 ? leftSectionSlider
                 : (location==='right-section-slider')
                 ? rightSectionSlider
                 : (location==='mode-toggle')
                 ? modeToggleCoords
                 : (location==='navbar')
                 ? navBarCoords
                 : defaultCoords;

    return (
        <div
            className={'highlighter ' + className}
            style={{
                left: coords.x,
                top: coords.y,
                width: coords.width,
                height: coords.height
            }}
        ></div>
    );
}
