// pages/NewWishPage.js
import React, { useState, useLayoutEffect, useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useForm } from 'react-hook-form'
import { useNavigate } from 'react-router'

import { ImageInput } from 'components/ImageInput'
import { TextInput } from 'components/TextInput'
import { SelectBox } from 'components/SelectBox'
import { StarRating } from 'components/StarRating'
import { ToggleInput } from 'components/ToggleInput'
import { Button } from 'components/Button'
import { IconButton } from 'components/IconButton'
import { generateId } from 'getters/idGen'
import { pushNewWish } from 'store/wishesSlice'
import { updateWishlistsWithNewWish } from 'store/wishlistSlice'

const defaultValues = {
    id: '',
    author: '',
    title: '',
    description: '',
    image: '',
    imageAspectRatio: '',
    external: '',
    price: '',
    currency: 'rouble',
    stars: 0,
    state: 'actual',
    wishlistIds: [],
}

export const NewWishPage = () => {
    // set page orientation:
    const [isLandscape, setIsLandscape] = useState(true)
    const setPageOrientation = () => setIsLandscape(window.innerWidth > 1280);

    useEffect(() => {
        setPageOrientation();
        window.addEventListener('resize', setPageOrientation)
        return () => window.removeEventListener('resize', setPageOrientation)
    },[]);

    // setting form:
    const { handleSubmit, register, setValue, reset, control } = useForm({ defaultValues });
    const currentUser = useSelector(state => state.users.current);
    const currentUserWishlists = useSelector(state => state.wishlists.ofCurrentUser);

    const statusOptions = [
        { value: 'actual', label: 'Актуально' },
        { value: 'completed', label: 'Исполнено' }
    ];
    const currencyOptions = [
        {value: 'rouble', label: '₽'},
        {value: 'euro', label: '€'},
        {value: 'dollar', label: '$'},
    ];
    const wishlistOptions = [
        ...currentUserWishlists.map(list => ({
                value: list.id,
                label: list.title
        }))
    ];

    //form methods:
    const dispatch = useDispatch();
    const navigate = useNavigate()

    const onSubmit = (data, e) => {
        console.log(data, e);
        dispatch(pushNewWish(data));
        dispatch(updateWishlistsWithNewWish(data));
    }
    const resetForm = () => {
        reset();
        imageInputRef.current.deleteImage();
    }
    
    // align labels:
    const [maxLabelWidth, setMaxLabelWidth] = useState(0);
    

    useEffect(() => {
        const labels = document.querySelectorAll('label.text');
        if (labels.length === 0) return;
        const labelWidths = [...labels].map(label => label.offsetWidth);
        setMaxLabelWidth(Math.max(...labelWidths));
    })

    const labelWidth = maxLabelWidth ? maxLabelWidth : null

    //set author & id fields:
    useLayoutEffect(() => {
        const id = generateId();
        setValue('id', id);
        setValue('author', currentUser);
    },[])

    const imageInputRef = useRef({})
    

    return (
        <div className='new-wish-page'>
            <form
                className={ isLandscape ? 'landscape' : 'portrait' }
                onSubmit={handleSubmit(onSubmit)}
            >
                <input
                    className='invis'
                    type='text'
                    {...register('id')}
                />
                <input
                    className='invis'
                    type='text'
                    {...register('author')}
                />
                <div className='left'>
                    <ImageInput
                        name='image'
                        register={register}
                        setValue={setValue}
                        imageInputRef={imageInputRef}
                    />
                </div>
                <div className='right'>
                    <TextInput
                        name='title'
                        register={register}
                        placeholder='Таинственный артефакт'
                        label='Название'
                        labelWidth={labelWidth}
                        required 
                    />
                    <TextInput
                        name='description'
                        register={register}
                        placeholder='Отметьте моменты, которые вам важны. Чем точнее вы опишете желание, тем проще друзьям будет его исполнить'
                        label='Описание'
                        labelWidth={labelWidth}
                        multiline
                        
                    />
                    <TextInput
                        name='external'
                        register={register}
                        placeholder='https://magicstore.com/goods/mysterious-artefact'
                        label='Ссылка'
                        labelWidth={labelWidth}
                    />
                    <div className='line-container'>
                        <TextInput
                            name='price'
                            register={register}
                            placeholder='4200'
                            label='Цена'
                            labelWidth={labelWidth}
                        />
                        <ToggleInput
                            name='currency'
                            control={control}
                            options={currencyOptions}
                        />
                    </div>
                    <div className='line-container'>
                        <SelectBox
                            name='state'
                            control={control}
                            options={statusOptions}
                            placeholder='Актуально'
                            label='Статус'
                            labelWidth={labelWidth}
                        />
                        <StarRating
                            name='stars'
                            control={control}
                            maxStars={3}
                            starBoxSize={4}
                        />
                    </div>
                    <SelectBox
                        name='wishlistIds'
                        control={control}
                        isMulti
                        options={wishlistOptions}
                        placeholder='Можно сразу добавить в вишлисты'
                        label='Вишлисты'
                        labelWidth={labelWidth}
                    />
                    <div className='line-container btn-group'>
                        <IconButton
                            icon='clear'
                            onClick={() => resetForm()}
                        />
                        <IconButton
                            icon='cancel'
                            onClick={() => navigate(-1)}
                        />
                        <Button
                            type='submit'
                            kind='primary'
                            text='Сохранить желание'
                            leftIcon='ok'
                            round
                        />
                    </div>
                </div>
            </form>
        </div>
    );
}
