// pages/SingleWishPage.js
import { React, useEffect } from 'react'
import { Link, useLocation } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'

import { getWishById, getWishlistById } from 'getters'
import { Icon, WishpageDescriptionPointer } from 'components/Icon'
import { User } from 'components/User'
import { Button } from 'components/Button'
import { setDescriptionPointerCoords } from 'store/highlighterSlice'

export const SingleWishPage = () => {
    const loca = useLocation().pathname;
    const currentWishId = loca.split('/').at(-1);
    const w = getWishById(currentWishId);

    const currency = w.currency==='rouble' ? ' ₽'
                   : w.currency==='yuan'   ? ' Ұ'
                   : w.currency==='usd'    ? ' $'
                   : w.currency==='euro'   ? ' €'
                   : '';
    
    const OuterLink = () => {
        if(!w.external) return null;
        const url = new URL(w.external)
        return (
            <a
                href={ w.external }
                className='inline-link'
            >
                { url.host }
                <Icon name='outerLink'/>
            </a>
        );
    }

    const WishlistOccurrences = () => {
        if(!w.wishlistIds.length) return (
            <div className='info-wishlists'>
                <span className='label'>Желание не включено в вишлист</span>
            </div>
        );

        const labelText = (w.wishlistIds.length == 1)
            ? 'Включено в вишлист'
            : 'Включено в вишлисты:'

        const mappedLinks = w.wishlistIds.map((id, index) => {
            const wl = getWishlistById(id);
            const comma = index ? (<span> , </span>) : null
            return (
                <div key={ index }>
                    { comma }
                    <Link
                        className='inline-link'
                        to={`/${ loca.split('/')[1] }/lists/${ wl.id }`}
                    >
                        { wl.title }
                    </Link>
                </div>
        )});

        return (
            <div className='info-field'>
                <span className='label'>{ labelText }</span>
                <div className='links'>{ mappedLinks }</div>
            </div>
    );}

    const UserInfo = () => {
        return (
            <div className='info-field'>
                <span className='label'>Желает</span>
                <User picSize='44' userName={ w.author }/>
            </div>
    )}

    const Description = () => {
        if(!w.description) return null;
        const offset = useSelector(state => state.highlighters.descriptionPointerCoords.x)
        const Pointer = () => {
            return (
                <div
                    className='description-pointer'
                    style={{
                        position: 'relative',
                        left: offset,
                    }}
                >
                    <WishpageDescriptionPointer/>
                </div>
        )}

        return (
            <div className='description'>
                <Pointer/>
                <div
                    className='container'
                    style={{minWidth: `${offset + 77}px`}}
                >{ w.description }</div>
            </div>
    )}

    const PriceLine = () => ( w.price
        ? <span className='price'>{ w.price + currency}</span>
        : null
    )

    const PrimaryButton = () => {
        const isMyWish = loca.split('/')[1] === 'my-wishes'
            ? true : false

        const content = (w.state === 'manifested' && isMyWish)
            ? {
                icon: 'ok',
                text: 'Отметить исполненным',
                handleClick() {console.log('жмяк')},
            }
            : (w.state === 'manifested' && !isMyWish)
            ? {
                icon: 'present',
                text: 'Исполнить желание',
                handleClick() {console.log('жмяк')},
            }
            : (w.state === 'reserved')
            ? {
                icon: 'lock',
                text: 'Исполнить желание',                
                isDisabled: true,
            }
            : (w.state === 'completed')
            ? {
                icon: 'ok',
                text: 'Желание исполнено',
                isDisabled: true,
            }
            : undefined

        if(!content) return null;

        return (
            <Button
                type='primary'
                leftIcon={ content.icon }
                text={ content.text }
                isDisabled={ !!content.isDisabled }
                handleClick={ content.handleClick }
    />);}

    const dispatch = useDispatch()
    useEffect(() => {
        dispatch(setDescriptionPointerCoords())
    },[])

    return (
        <div className='wish-page'>
            <div className='img-block'>
                <img src={ w.image }></img>
            </div>
            <div className='content-block'>
                <div className='header'>
                    <span className='title'>{ w.title }</span>
                    <OuterLink/>
                </div>
                <div className='info'>
                    <WishlistOccurrences/>
                    <UserInfo/>
                    <Description/>
                </div>
                <div className='actions'>
                    <PriceLine/>
                    <PrimaryButton/>
                </div>
            </div>
        </div>
);}
