// store/highlighterSlice.js
import { createSlice, current } from '@reduxjs/toolkit'

const highlighterSlice = createSlice({
    name: 'highLighter',
    initialState: {
        inAppHeaderCoords: {
            x: 0,
            y: 36,
            width: 0,
            height: 11,
        },
        inModeToggleCoords: {
            x: 3,
            y: 3,
            width: 36,
            height: 36,
        },
        inNavBarCoords: {
            x: 27,
            y: 36,
            width: 114,
            height: 6,
            last: undefined,
        },
        descriptionPointerCoords: {
            x: 69,
        },
    },
    reducers: {
        highlightSectionStart(state) {
            const myWishesTab = document.querySelector('.section.my-wishes');
            const myInvitesTab = document.querySelector('.section.my-invites');
            const logo = document.querySelector('.logo-btn');
            const logoCoords = logo.getBoundingClientRect();
            const highlighter = document.querySelector('.goo-highlighters .highlighter');
            highlighter.classList.add('animated');
            const dotSize = 44;
            state.inAppHeaderCoords.x = logoCoords.x + logoCoords.width / 2 - dotSize / 2;
            state.inAppHeaderCoords.y = logoCoords.y + logoCoords.height / 2 - dotSize / 2 + 6;
            state.inAppHeaderCoords.width = state.inAppHeaderCoords.height = dotSize;
            console.log('SWITCH SECTION FIRED');
            return state;
        },
        highlightSectionFinish(state) {
            const activeSection = document.querySelector('.section.active');
            if(!activeSection) return state;

            const activeTabCoords = activeSection.getBoundingClientRect();

            state.inAppHeaderCoords.x = Math.floor(activeTabCoords.x);
            state.inAppHeaderCoords.y = Math.floor(activeTabCoords.y) + 36;
            state.inAppHeaderCoords.width = Math.ceil(activeTabCoords.width);
            state.inAppHeaderCoords.height = 11;
        },
        setHighlighterCoordsInModeToggle(state) {
            const activeModeButton = document.querySelector('.toggle-button.active');
            if(!activeModeButton) return state;

            state.inModeToggleCoords.x = activeModeButton.offsetLeft + 3;
        },
        setHighlighterCoordsInNavBar(state) {
            const next = document.querySelector('.nav.option.active');
            const last = current(state).inNavBarCoords.last;
            if(!next) return state;

            if(!last) {
                state.inNavBarCoords.x = next.offsetLeft + 16;
                state.inNavBarCoords.width = next.offsetWidth - 33;
            }

            if(last && next.offsetLeft >= last.offsetLeft) {
                state.inNavBarCoords.width = next.offsetLeft - last.offsetLeft + next.offsetWidth - 33;
            }

            if(last && next.offsetLeft < last.offsetLeft) {
                state.inNavBarCoords.x = next.offsetLeft + 16;
                state.inNavBarCoords.width = last.offsetWidth + last.offsetLeft - next.offsetLeft - 33;
            }

            state.inNavBarCoords.last = {
                offsetLeft: next.offsetLeft,
                offsetWidth: next.offsetWidth,
            };
        },
        animationSecondStep(state) {
            const coords = document.querySelector('.nav.option.active');
            if(!coords) return state;
            state.inNavBarCoords.x = coords.offsetLeft + 16;
            state.inNavBarCoords.width = coords.offsetWidth - 33;
        },
        setDescriptionPointerCoords(state) {
            const offset = document.querySelector('.info-field .user').offsetLeft - 11;
            if(!offset) return state;

            state.descriptionPointerCoords.x = offset;
        }
    },
});

export const {
    highlightSectionStart,
    highlightSectionFinish,
    setHighlighterCoordsInModeToggle,
    setHighlighterCoordsInNavBar,
    animationSecondStep,
    setDescriptionPointerCoords,
} = highlighterSlice.actions;

export default highlighterSlice.reducer;
