// store/highlighterSlice.js
import { createSlice } from '@reduxjs/toolkit'

const highlighterSlice = createSlice({
    name: 'highLighter',
    initialState: {
        inAppHeaderCoords: {
            x: 0,
            y: 0,
            width: 0,
            height: 11,
        },
        inModeToggleCoords: {
            x: 3,
            y: 3,
            width: 36,
            height: 36,
        },
        inNavBarCoords: {
            x: 0,
            y: 35,
            width: 33,
            height: 6,
        }
    },
    reducers: {
        setHighlighterCoordsInAppHeader(state) {
            const activeTab = document.querySelector('.header-nav.active');
            if(!activeTab) return state;

            const activeTabCoords = activeTab.getBoundingClientRect();

            state.inAppHeaderCoords.x = Math.floor(activeTabCoords.x);
            state.inAppHeaderCoords.y = Math.floor(activeTabCoords.y) + 36;
            state.inAppHeaderCoords.width = Math.ceil(activeTabCoords.width);
        },
        setHighlighterCoordsInModeToggle(state) {
            const activeModeButton = document.querySelector('.toggle-button.active');
            if(!activeModeButton) return state;

            state.inModeToggleCoords.x = activeModeButton.offsetLeft + 3;
        },
        setHighlighterCoordsInNavBar(state) {
            const activeOption = document.querySelector('.nav.option.active');
            if(!activeOption) return state;
            
            state.inNavBarCoords.x = activeOption.offsetLeft + 16;
            state.inNavBarCoords.width = activeOption.offsetWidth - 33;
        },
    },
});

export const {
    setHighlighterCoordsInAppHeader,
    setHighlighterCoordsInModeToggle,
    setHighlighterCoordsInNavBar,
} = highlighterSlice.actions;

export default highlighterSlice.reducer;
