// store/historySlice.js
import { createSlice } from '@reduxjs/toolkit'

const historySlice = createSlice({
    name: 'history',
    initialState: {
        myWishesSection: {
            itemsModeLast: '/my-wishes/items/actual',
            listsModeLast: '/my-wishes/lists',
            last: '/my-wishes/items/actual',
        },
        myInvitesSection: {
            itemsModeLast: '/my-invites/items/reserved',
            listsModeLast: '/my-invites/lists',
            last: '/my-invites/items/reserved',
        },
    },
    reducers: {
        updateHistory(state,{ payload }) {
            const section = payload.split('/').at(1);
            const mode = payload.split('/').at(2);
            const tab = payload.split('/').at(3);
            
            if(tab === 'new') return state;

            switch (section) {
                case 'my-wishes':
                    state.myWishesSection.last = payload;
                    switch (mode) {
                        case 'items':
                            state.myWishesSection.itemsModeLast = payload;
                            break;

                        case 'lists':
                            state.myWishesSection.listsModeLast = payload;
                            break;
                    };
                    break;

                case 'my-invites':
                    state.myInvitesSection.last = payload;
                    switch (mode) {
                        case 'items':
                            state.myInvitesSection.itemsModeLast = payload;
                            break;

                        case 'lists':
                            state.myInvitesSection.listsModeLast = payload;
                            break;
                    };
                    break;
                default: return state;
            }
        }
    },
});

export const {
    updateHistory,
} = historySlice.actions;

export default historySlice.reducer;
