// store/sliderSlice.js
import { createSlice, current } from '@reduxjs/toolkit'

const sliderSlice = createSlice({
    name: 'sliders',
    initialState: {
        leftSectionSlider: {
            x: -159,
            y: 37,
            width: 148,
            height: 11,
        },
        rightSectionSlider: {
            x: 209,
            y: 37,
            width: 151,
            height: 11,
        },
    },
    reducers: {
        switchSectionSlider(state) {
            console.log('WE ARE HERE YEAHHH', current(state))
            const activeSection = document.querySelector('.section.active');
            if(!activeSection) return state;
            
            const rightSection = document.querySelector('.section.my-invites');
            const switchToRight = activeSection === rightSection;

            const activeSectionCoords = activeSection.getBoundingClientRect();
            const logoCoords = document.querySelector('.logo-btn').getBoundingClientRect();
            const dotSize = 44;

            if(switchToRight) {
                state.leftSectionSlider = {
                    x: dotSize / 2,
                    y: Math.round(logoCoords.height / 2 - dotSize / 2 + 6),
                    width: dotSize,
                    height: dotSize,
                };
                state.rightSectionSlider = {
                    x: Math.round(logoCoords.width + 11),
                    y: Math.round(activeSectionCoords.y + 27),
                    width: Math.round(activeSectionCoords.width),
                    height: 11,
                };
            } else {
                state.leftSectionSlider = {
                    x: Math.round(- activeSectionCoords.width - 11),
                    y: Math.round(activeSectionCoords.y + 27),
                    width: Math.round(activeSectionCoords.width),
                    height: 11,
                };
                state.rightSectionSlider = {
                    x: Math.round(logoCoords.width - 3 * dotSize / 2),
                    y: Math.round(logoCoords.height / 2 - dotSize / 2 + 2),
                    width: dotSize,
                    height: dotSize,
                };
            }
        },
    }
});

export const {
    switchSectionSlider,
} = sliderSlice.actions;

export default sliderSlice.reducer;
