// store/wishesSlice.js
import { createSlice, createAsyncThunk, current } from '@reduxjs/toolkit'

export const fetchWishes = createAsyncThunk(
    'wishes/fetchWishes',
    async function() {
        const response = await fetch('https://run.mocky.io/v3/03933f0e-09c4-4f99-9b67-e2b7dd8bd093');
        const data = await response.json();
        return data;
    }
);

const wishesSlice = createSlice({
    name: 'wishes',
    initialState: {
        all: [],
        status: null,
        error: null,
        ofCurrentUser: [],
        ofOtherUsers: [],
    },
    reducers: {
        filterWishesByUser(state,{ payload }) {
            // payload: userName
            if(!current(state).all.length) return state;

            state.ofCurrentUser = state.all.filter(
                wish => wish.author === payload
            );
            state.ofOtherUsers = state.all.filter(
                wish => wish.author !== payload
            );
        },
        pushNewWish(state,{ payload }) {
            // payload: {wish}
            state.all.push(payload);
            state.ofCurrentUser.push(payload);
        },
        updateWishesWithNewWishlist(state,{ payload }) {
            // payload: {wishlist}
            state
                .all
                .filter(wish => payload.wishIds.includes(wish.id))
                .forEach(wish => wish.wishlistIds.push(payload.id));
            state
                .ofCurrentUser
                .filter(wish => payload.wishIds.includes(wish.id))
                .forEach(wish => wish.wishlistIds.push(payload.id));
        }
    },
    extraReducers: {
        [fetchWishes.pending]: (state) => {
            state.status = 'loading';
            state.error = null;
        },
        [fetchWishes.fulfilled]: (state,{ payload }) => {
            state.status = 'resolved';
            state.all = payload;
        },
        [fetchWishes.rejected]: (state, action) => {},
    }
});

export const {
    filterWishesByUser,
    pushNewWish,
    updateWishesWithNewWishlist
} = wishesSlice.actions;

export default wishesSlice.reducer;
