// store/wishColumnSlice.js
import { createSlice, current } from '@reduxjs/toolkit'

const wishColumnSlice = createSlice({
    name: 'wishColumns',
    initialState: {
        wishColumns: [
            {
                id: 0,
                wishIds: [],
                height: 0,
            }
        ]
    },
    reducers: {
        addColumn(state, action) {
            //console.log('Колонок до: ', state.wishColumns.length);

            state.wishColumns.push({
                id: state.wishColumns.length,
                wishIds: [],
                height: 0,
            });

            //console.log('Колонок после: ', state.wishColumns.length);
            console.log(current(state))
        },
        removeColumn(state, action) {},
        addWish(state, action) {},
        updateHeight(state, action) {},
        clearColumns(state, action) {},
    }
})

export const {
    addColumn,
    removeColumn,
    addWish,
    updateHeight,
    clearColumns
} = wishColumnSlice.actions;

export default wishColumnSlice.reducer;


// Cycle for generate columns

/*
import { useDispatch } from 'react-redux'

import { addColumn } from 'store/wishColumnSlice'


const dispatch = useDispatch();
const plusColumn = () => dispatch(addColumn());

for (var i=0; i<columnsQty; i++) {
    plusColumn()
}
*/
