// store/wishListSlice.js
import { createSlice, current } from '@reduxjs/toolkit'

import { wishlists } from 'api/mockWishlists'

const wishlistSlice = createSlice({
    name: 'wishlists',
    initialState: {
        all: wishlists,
        ofCurrentUser: [],
        ofOtherUsers: [],
    },
    reducers: {
        filterWishlistsByUser(state,{ payload }) {
            // payload: userName
            if(!current(state).all.length) return state;
            
            state.ofCurrentUser = current(state).all.filter(
                wishlist => wishlist.author === payload
            );
            state.ofOtherUsers = current(state).all.filter(
                wishlist => wishlist.author !== payload
            );
        },
        updateWishlistsWithNewWish(state,{ payload }) {
            // payload: {wish}
            state
                .all
                .filter(
                    list => payload.wishlistIds.includes(list.id)
                ).forEach(
                    list => list.wishIds.push(payload.id)
                );
            state
                .ofCurrentUser
                .filter(
                    list => payload.wishlistIds.includes(list.id)
                ).forEach(
                    list => list.wishIds.push(payload.id)
                );
        }
    }
});

export const { filterWishlistsByUser, updateWishlistsWithNewWish } = wishlistSlice.actions;

export default wishlistSlice.reducer;
