// store/wishesSlice.js
import { createSlice, createAsyncThunk, current } from '@reduxjs/toolkit'

export const fetchWishes = createAsyncThunk(
    'wishes/fetchWishes',
    async function() {
        const response = await fetch('https://run.mocky.io/v3/f5dae5ef-63d8-4003-91e9-33fff4630d64');
        const data = await response.json();
        return data;
    }
);

const wishesSlice = createSlice({
    name: 'wishes',
    initialState: {
        all: [],
        status: null,
        error: null,
        ofCurrentUser: [],
        ofOtherUsers: [],
    },
    reducers: {
        filterWishesByUser(state,{ payload }) {
            console.log('filtration by user starts with: ', payload, current(state).all)
            if(!current(state).all.length) return state;

            state.ofCurrentUser = state.all.filter(
                wish => wish.author === payload
            );
            state.ofOtherUsers = state.all.filter(
                wish => wish.author !== payload
            );
        },
    },
    extraReducers: {
        [fetchWishes.pending]: (state) => {
            state.status = 'loading';
            state.error = null;
        },
        [fetchWishes.fulfilled]: (state,{ payload }) => {
            state.status = 'resolved';
            state.all = payload;
        },
        [fetchWishes.rejected]: (state, action) => {},
    }
});

export const { filterWishesByUser } = wishesSlice.actions;

export default wishesSlice.reducer;
