// store/wishColumnSlice.js
import { createSlice, current } from '@reduxjs/toolkit'

const columnsInitial = []

const wishColumnSlice = createSlice({
    name: 'wishColumns',
    initialState: {
        wishColumns: columnsInitial,
        workSpaceWidth: 0,
        wishesToRender: [],
    },
    reducers: {
        setWorkSpaceWidth(state,{ payload }) {
            console.log('work-space width is: ', payload);
            state.workSpaceWidth = payload;
        },
        addColumn(state) {
            console.log(' wishesToRender: ', current(state.wishesToRender).length);
            if (current(state.wishesToRender).length) {
                state.wishColumns.push({
                    id: state.wishColumns.length,
                    wishes: [current(state.wishesToRender.shift())]
                });
            } else return
        },
        resetColumns(state) {
            console.log('resetColumns fired');
            state.wishColumns = columnsInitial;
        },
        removeColumn(state) {
            state.wishColumns.pop();
        },
        fillWaitList(state,{ payload }) {
            console.log('fillWaitList fired with: ', payload);
            state.wishesToRender = [...payload];
        },
        takeWishToRender(state,{ payload }) {
            console.log('список к рендеру: ', current(state.wishesToRender), 'takeWishToRender fired with: ', payload);
            const wish = state.wishesToRender.shift();
            wish && state.wishColumns[ payload ].wishes.push( wish );
        }

    }
})

export const {
    setWorkSpaceWidth,
    addColumn,
    resetColumns,
    removeColumn,
    fillWaitList,
    popWaitList,
    addWishToColumn,
    takeWishToRender,
} = wishColumnSlice.actions;

export default wishColumnSlice.reducer;
