// store/wishColumnSlice.js
import { createSlice, current } from '@reduxjs/toolkit'

const columnsInitial = [{
    id: 0,
    wishes: []
}]

const wishColumnSlice = createSlice({
    name: 'wishColumns',
    initialState: {
        wishColumns: columnsInitial,
        workSpaceWidth: 0,
        wishesToRender: [],
    },
    reducers: {
        setWorkSpaceWidth(state,{ payload }) {
            state.workSpaceWidth = payload;
        },
        addColumn(state) {
            state.wishColumns.push({
                id: state.wishColumns.length,
                wishes: []
            });
            console.log('current columns: ', state.wishColumns.length)
        },
        resetColumns(state) {
            state.wishColumns = columnsInitial;
        },
        removeColumn(state) {
            state.wishColumns.pop();
        },
        fillWaitList(state,{ payload }) {
            console.log('До сюда дошли', payload);
            state.wishesToRender = [...payload];
        },
        takeWishToRender(state,{ payload }) {
            console.log('список к рендеру: ', current(state.wishColumns[payload].wishes), 'payload: ', payload);
            const wish = state.wishesToRender.shift();
            console.log('wish is: ', wish, 'current wait-list: ', current(state.wishesToRender))
            state.wishColumns[ payload ].wishes.push( wish );
        }

    }
})

export const {
    setWorkSpaceWidth,
    addColumn,
    resetColumns,
    removeColumn,
    fillWaitList,
    popWaitList,
    addWishToColumn,
    takeWishToRender,
} = wishColumnSlice.actions;

export default wishColumnSlice.reducer;
