// store/wishesSlice.js
import { createSlice, createAsyncThunk, current } from '@reduxjs/toolkit'

export const fetchWishes = createAsyncThunk(
    'wishes/fetchWishes',
    async function() {
        const response = await fetch('https://run.mocky.io/v3/f8cd68e2-3f87-4fc4-b3d2-c4bec7d43605');
        const data = await response.json();
        return data;
    }
);

const wishesSlice = createSlice({
    name: 'wishes',
    initialState: {
        all: [],
        status: null,
        error: null,
        filteredWishes: [],
    },
    reducers: {
        filterWishes(state) {
            state.filteredWishes = [...wishes];
        },
    },
    extraReducers: {
        [fetchWishes.pending]: (state) => {
            state.status = 'loading';
            state.error = null;
        },
        [fetchWishes.fulfilled]: (state,{ payload }) => {
            state.status = 'resolved';
            state.all = payload;
        },
        [fetchWishes.rejected]: (state, action) => {},
    }
});

export const { filterWishes } = wishesSlice.actions;

export default wishesSlice.reducer;
