// api/mockWishes.js
export const wishes = [
    {
      id: "1",
      name: "Очки калейдоскопы",
      description:
        "Те что на фото мне нравятся больше других на алике, но я готов довериться вашему вкусу =)",
      image: "/images/caleidoscope_glasses.jpg",
      external:
        "https://aliexpress.ru/item/32857629916.html?sku_id=12000027017624705&spm=a2g0o.search.0.0.3cb86209B3017l",
      circa: true,
      price: 400,
      currency: "rouble",
      stars: 3,
      state: "manifested",
      authorId: "5",
    },
    {
      id: "2",
      name: "Канджира",
      description:
        "(Если вы разбираетесь) С кожей ящерицы, профессиональный инструмент.",
      image: "/images/kanjira.jpg",
      external: "",
      circa: true,
      price: 20000,
      currency: "rouble",
      stars: 1,
      state: "reserved",
      authorId: "5",
    },
    {
      id: "3",
      name: "Столик для ноутбука",
      description:
        "С подставкой для мыши. Чтобы работать в красивых местах, где солнце и люди улыбаются, а не вот это вот всё >.<",
      image: "/images/laptop-table.jpg",
      external:
        "https://markakachestva.ru/rating-of/4126-luchshie-stoliki-dlja-noutbuka.html",
      circa: true,
      price: 2000,
      currency: "rouble",
      stars: 0,
      state: "manifested",
      authorId: "5",
    },
    {
      id: "4",
      name: "Pixel-poi HD",
      description:
        "ePoi PRO 180HD или похожие",
      image: "/images/pixel-poi.jpg",
      external:
        "https://epoi.ru/epoi-180hd/",
      circa: true,
      price: 62000,
      currency: "rouble",
      stars: 3,
      state: "manifested",
      authorId: "5",
    },
    {
      id: "5",
      name: "Книга Чалмерса “Reality +”",
      description:
        "Перевода вроде нет, можно на английском =)",
      image: "/images/reality+.jpg",
      external: null,
      circa: false,
      price: null,
      currency: "",
      stars: 2,
      state: "reserved",
      authorId: "5",
    },
    {
      id: "6",
      name: "Роуп дарт огненный",
      description:
        "Точно такой как по ссылке",
      image: "/images/rope-dart.jpg",
      external:
        "https://vk.com/market-9733488?w=product-9733488_5183097%2Fquery",
      circa: false,
      price: 3990,
      currency: "rouble",
      stars: 2,
      state: "manifested",
      authorId: "5",
    },
    {
      id: "7",
      name: "Огненный стафф 100см",
      description:
        "По ссылке или другой, главное чтобы разбирался на три части и был ровным — для изоляции",
      image: "/images/staff.jpg",
      external:
        "https://jugglingstore.ru/ognennyj-staff-dlya-faer-shou/1374-ognennyy-staff-nybras-100-sm-spinning-and-isolation-fire-staff.html",
      circa: false,
      price: 5990,
      currency: "rouble",
      stars: 1,
      state: "manifested",
      authorId: "5",
    },
    {
      id: "8",
      name: "Mi Smart Band 5",
      description:
        "Фитнес браслет со сменными ремешками. Цвет чёрный или серый",
      image: "/images/MiSmartBand5.jpg",
      external: null,
      circa: true,
      price: 3000,
      currency: "rouble",
      stars: 3,
      state: "manifested",
      authorId: "1",
    },
    {
      id: "9",
      name: "JBL Charge 5",
      description: null,
      image: "/images/JBLCharge5.jpg",
      external: null,
      circa: true,
      price: 23000,
      currency: "rouble",
      stars: 3,
      state: "manifested",
      authorId: "4",
    },
    {
      id: "10",
      name: "Сертификат в спа-салон",
      description: null,
      image: "/images/spaSertificate.jpg",
      external: null,
      circa: false,
      price: null,
      currency: "rouble",
      stars: 1,
      state: "manifested",
      authorId: "3",
    },
    {
      id: "11",
      name: "Игрушка для кошки",
      description: null,
      image: "/images/kittenToy.jpg",
      external:
        "https://4lapy.ru/catalog/koshki/igrushki-koshki/igrushka-dlya-koshek-draznilka-40-sm.html?offer=128611",
      circa: false,
      price: 379,
      currency: "rouble",
      stars: 1,
      state: "manifested",
      authorId: "8",
    },
    {
      id: "12",
      name: "Подушка-единорожка",
      description: null,
      image: "/images/unicorn.jpg",
      external:
        "https://www.ozon.ru/product/podushka-dekorativnaya-edinorog-206029810/?asb=XA1nLYKW1OuS4b3B%252FBdqVcRRjyz6WvM9Mq4Qo3MZCzo%253D&asb2=zO7om_ebjdCJFQKeqRet2hR3Go20bNOQQJCStUQJLPKaPUTOdUjnN-ijX2fWHzbE&keywords=%D0%BF%D0%BE%D0%B4%D1%83%D1%88%D0%BA%D0%B0+%D0%B5%D0%B4%D0%B8%D0%BD%D0%BE%D1%80%D0%BE%D0%B3&sh=URA_ogAAAA",
      circa: false,
      price: 1096,
      currency: "rouble",
      stars: 2,
      state: "manifested",
      authorId: "2",
    },
    {
      id: "13",
      name: "Блендер",
      description:
        "Redmond RHB-2981",
      image: "/images/blender.jpg",
      external: null,
      circa: false,
      price: 3499,
      currency: "rouble",
      stars: 0,
      state: "manifested",
      authorId: "9",
    },
    {
      id: "14",
      name: "Чайник",
      description:
        "Что-нибудь в этом стиле",
      image: "/images/cuttle.jpg",
      external: null,
      circa: true,
      price: 4000,
      currency: "rouble",
      stars: 2,
      state: "manifested",
      authorId: "6",
    },
    {
      id: "15",
      name: "Аккумуляторный сенсорный светильник «Луна»",
      description:
        "«Луна» - идеальный подарок и отличное украшение, порадуйте себя или близких с этим теплым очаровательным светильником.",
      image: "/images/moonLight.jpg",
      external:
        "https://www.millionpodarkov.ru/catalog/presents_6485429.htm#tovar",
      circa: false,
      price: 1590,
      currency: "rouble",
      stars: 3,
      state: "manifested",
      authorId: "7",
    },
    {
      id: "16",
      name: "3D-конструктор «МАСКА ЗАЯЦ розовый»",
      description:
        "3D-конструктор PAPERRAZ для сборки полигональной модели «МАСКА ЗАЯЦ розовый» из дизайнерского картона (черный)",
      image: "/images/rabbitMask.jpg",
      external:
        "https://www.millionpodarkov.ru/catalog/presents_8024583.htm#tovar",
      circa: false,
      price: 990,
      currency: "rouble",
      stars: 2,
      state: "manifested",
      authorId: "7",
    },
    {
      id: "17",
      name: "Набор для рисования по номерам Милые котята",
      description:
        "Холст на подрамнике и пронумерованные краски",
      image: "/images/artKittens.jpg",
      external:
        "https://www.millionpodarkov.ru/catalog/presents_7999929.htm#tovar",
      circa: false,
      price: 490,
      currency: "rouble",
      stars: 0,
      state: "manifested",
      authorId: "7",
    },
    {
      id: "18",
      name: "FurReal Friends Тигренок на поводке",
      description: null,
      image: "/images/tiger.jpg",
      external:
        "https://www.detmir.ru/product/index/id/3195482/",
      circa: true,
      price: 1000,
      currency: "rouble",
      stars: 1,
      state: "manifested",
      authorId: "7",
    },
    {
      id: "19",
      name: "Игрушка-обнимашка Кот-батон",
      description: null,
      image: "/images/cat.jpg",
      external: null,
      circa: false,
      price: null,
      currency: "rouble",
      stars: 2,
      state: "manifested",
      authorId: "7",
    },
]
