// api/mockWishlists.js
export const wishlists = [
    {
        id: 'djVJJm',
        author: 'Neuroslav',
        wishIds: [
            'irM8c9',
            'NDWlMo',
            'RDLl3K',
            'gzu30w',
            'R7DjcR',
            'EeXy1l',
            '4wMdrK',
        ],
        title: 'День рождения',
        description: 'Привет, друг. Ты здесь, а это значит что я буду счастлив видеть тебя на вечеринке в честь моего дня рождения. Приходи со своей второй половинкой ;)',
        date: '20.04.2022',
    },
    {
        id: 'MxFLI5',
        author: 'Htonatan',
        wishIds: ['bgpcoX',],
        title: '👑 Head of department 👑',
        description: '',
        date: '12.03.2022',
    },
    {
        id: 'HdvrZs',
        author: 'ManneKumar',
        wishIds: ['OvlyHR',],
        title: 'My birthday',
        description: '',
        date: '15.04.2022',
    },
    {
        id: 'FSFYIh',
        author: 'SarraMarch',
        wishIds: [
            'RHASPg',
            'RDriQV',
            'CK2VzM',
            'oeCCjx',
            'PYvlnk',
        ],
        title: 'My 6th birthday!',
        description: '',
        date: '11.06.2022',
    },
    {
        id: 'lkCpxz',
        author: 'Jennishada',
        wishIds: ['HbciM4',],
        title: 'Me and Mike’s Wedding Day',
        description: '',
        date: '30.05.2021',
    },
    {
        id: 'D0Gzii',
        author: 'ToriahaliYulandi',
        wishIds: ['OvLh09',],
        title: 'А у Мурки котята родились! :3 ',
        description: 'Приходите гладить пушистиков!',
        date: '20.07.2022',
    },
    {
        id: 'qMuBSN',
        author: 'Yinalin',
        wishIds: ['zLhS69',],
        title: '33! 🎂 ',
        description: '',
        date: '10.10.2021',
    },
    {
        id: '3G5gS5',
        author: 'JariSmith',
        wishIds: ['quiZvc',],
        title: 'Farewell party',
        description: '',
        date: '04.11.2021',
    },
    {
        id: 'moTu2H',
        author: 'Rakeishand',
        wishIds: ['IitA1I',],
        title: 'Housewarming party',
        description: '',
        date: '01.02.2022',
    },
]
