// api/mockWishlists.js
export const wishlists = [
    {
        id: 'hmT6kL',
        author: 'Neuroslav',
        wishIds: [
            'UYvzGe',
            'ughoKY',
            'F21MG9',
            'eF8bWu',
            '1MAk7o',
        ],
        title: 'День рождения',
        description: 'Привет, друг. Ты здесь, а это значит что я буду счастлив видеть тебя на вечеринке в честь моего дня рождения. Приходи со своей второй половинкой ;)',
        date: ['11', '06', '2022'],
    },
    {
        id: 'djVJJm',
        author: 'Neuroslav',
        wishIds: [
            'irM8c9',
            'NDWlMo',
            'RDLl3K',
            'gzu30w',
            'R7DjcR',
            'EeXy1l',
            '4wMdrK',
        ],
        title: 'Улетаем на зимовку, приезжайте повидаться',
        description: 'Приходите, не скоро увидимся) кто придёт — тому привезём гостинцев 😉',
        date: ['20', '09', '2022'],
    },
    {
        id: 'MxFLI5',
        author: 'Htonatan',
        wishIds: ['bgpcoX',],
        title: '👑 Head of department 👑',
        description: "Yo brooo who's your daddy!?",
        date: ['12', '07', '2022'],
    },
    {
        id: 'HdvrZs',
        author: 'ManneKumar',
        wishIds: ['OvlyHR',],
        title: 'My birthday',
        description: '',
        date: ['15', '08', '2022'],
    },
    {
        id: 'FSFYIh',
        author: 'SarraMarch',
        wishIds: [
            'RHASPg',
            'RDriQV',
            'CK2VzM',
            'oeCCjx',
            'PYvlnk',
        ],
        title: 'My 6th birthday!',
        description: '',
        date: ['16', '05', '2022'],
    },
    {
        id: 'lkCpxz',
        author: 'Jennishada',
        wishIds: ['HbciM4',],
        title: 'Свадьба и венчание',
        description: 'Ураааа!!! Мы женимся! =))) приглашаем всех в ЗАГС-ресторан-клуб. Самые стойкие поедут на афтепати  😏',
        date: ['30', '05', '2021'],
    },
    {
        id: 'D0Gzii',
        author: 'ToriahaliYulandi',
        wishIds: ['OvLh09',],
        title: 'А у Мурки котята родились! :3 ',
        description: 'Приходите гладить пушистиков!',
        date: ['25', '06', '2022'],
    },
    {
        id: 'qMuBSN',
        author: 'Yinalin',
        wishIds: ['zLhS69',],
        title: '33! 🎂 ',
        description: '',
        date: ['10', '10', '2021'],
    },
    {
        id: '3G5gS5',
        author: 'JariSmith',
        wishIds: ['quiZvc',],
        title: 'Прощальная вечеринка',
        description: 'Мы с родителями уезжаем в другой город. Буду по вам всем очень скучать! 🤧 ',
        date: ['04', '11', '2021'],
    },
    {
        id: 'moTu2H',
        author: 'Rakeishand',
        wishIds: ['IitA1I',],
        title: 'Новоселье',
        description: 'О-хо-хо, друзья, мы наконец закончили ремонт! Милости просим в наше с  @Yinalin новое гнёздышко',
        date: ['01', '02', '2022'],
    },
]
