// api/mockWishlists.js
export const wishLists = [
    {
        id: '1',
        authorId: '5',
        wishIds: [],
        title: 'День рождения',
        description: 
            'Привет, друг. Ты здесь, а это значит что я буду счастлив видеть тебя на вечеринке в честь моего дня рождения. Приходи со своей второй половинкой ;)',
        date: '20.04.2022',
    },
    {
        id: '2',
        authorId: '1',
        wishIds: [],
        title: '👑 Head of department 👑',
        description: '',
        date: '12.03.2022',
    },
    {
        id: '3',
        authorId: '4',
        wishIds: [],
        title: 'My birthday',
        description: '',
        date: '15.04.2022',
    },
    {
        id: '4',
        authorId: '7',
        wishIds: [],
        title: 'My 6th birthday!',
        description: '',
        date: '11.06.2022',
    },
    {
        id: '5',
        authorId: '3',
        wishIds: [],
        title: 'Me and Mike’s Wedding Day',
        description: '',
        date: '30.05.2021',
    },
    {
        id: '6',
        authorId: '8',
        wishIds: [],
        title: 'My cat gave birth to kittens! :3 ',
        description: '',
        date: '20.07.2021',
    },
    {
        id: '7',
        authorId: '9',
        wishIds: [],
        title: '33! 🎂 ',
        description: '',
        date: '10.10.2021',
    },
    {
        id: '8',
        authorId: '2',
        wishIds: [],
        title: 'Farewell party',
        description: '',
        date: '04.11.2021',
    },
    {
        id: '9',
        authorId: '6',
        wishIds: [],
        title: 'Housewarming party',
        description: '',
        date: '01.02.2022',
    },
]
