// api/mockUsers.js
export const users = [
    {
        userName: 'Htonatan',
        email: 'htonatan-the-best@wisssh.com',
        password: 'blackmamba',
        avatar: '/avatars/Htonatan.jpg',
        wishIds: ['bgpcoX'],
        wishlistIds: ['MxFLI5'],
        invites: [],
        reservedPresents: [],
    },
    {
        userName: 'JariSmith',
        email: 'jarishka@wisssh.com',
        password: 'burnmotherfucker',
        avatar: '/avatars/Jari Smith.jpg',
        wishIds: ['quiZvc'],
        wishlistIds: ['3G5gS5'],
        invites: [],
        reservedPresents: [],
    },
    {
        userName: 'Jennishada',
        email: 'jennycutie@wisssh.com',
        password: 'notjenifer',
        avatar: '/avatars/Jennishada.jpg',
        wishIds: ['HbciM4'],
        wishlistIds: ['lkCpxz'],
        invites: ['djVJJm'],
        reservedPresents: [],
    },
    {
        userName: 'ManneKumar',
        email: 'mannekumar@wisssh.com',
        password: 'kumarkumar',
        avatar: '/avatars/Manne Kumar.jpg',
        wishIds: ['OvlyHR'],
        wishlistIds: ['HdvrZs'],
        invites: ['djVJJm'],
        reservedPresents: ['EeXy1l'],
    },
    {
        userName: 'Neuroslav',
        email: 'neuroslav1488@wisssh.com',
        password: 'uroborossiaforever',
        avatar: '/avatars/Neuroslav.jpg',
        wishIds: [
            'irM8c9',
            'NDWlMo',
            'RDLl3K',
            'gzu30w',
            'R7DjcR',
            'EeXy1l',
            '4wMdrK',
            'UYvzGe',
            'ughoKY',
            'F21MG9',
            'eF8bWu',
            '1MAk7o',
            'G1f82v',
            'kR19s0',
        ],
        wishlistIds: [
            'djVJJm',
            'hmT6kL'
        ],
        invites: [
            'MxFLI5',
            'HdvrZs',
            'FSFYIh',
            'lkCpxz',
            'D0Gzii',
            'qMuBSN',
            '3G5gS5',
            'moTu2H',
        ],
        reservedPresents: [
            'bgpcoX',
            'OvlyHR',
            'HbciM4',
            'OvLh09',
            'quiZvc',
            'zLhS69',
            'IitA1I',
            'RHASPg',
        ],
    },
    {
        userName: 'Rakeishand',
        email: 'rakeishand@wisssh.com',
        password: 'iamrakeishand',
        avatar: '/avatars/Rakeishand.jpg',
        wishIds: ['IitA1I'],
        wishlistIds: ['moTu2H'],
        invites: [],
        reservedPresents: [],
    },
    {
        userName: 'SarraMarch',
        email: 'sarra16@wisssh.com',
        password: 'mylittlepony',
        avatar: '/avatars/Sarra March.jpg',
        wishIds: [
            'RHASPg',
            'RDriQV',
            'CK2VzM',
            'oeCCjx',
            'PYvlnk',
        ],
        wishlistIds: ['FSFYIh'],
        invites: [],
        reservedPresents: [],
    },
    {
        userName: 'ToriahaliYulandi',
        email: 'toyu@wisssh.com',
        password: '6nh%k$/',
        avatar: '/avatars/Toriahali Yulandi.jpg',
        wishIds: ['OvLh09'],
        wishlistIds: ['D0Gzii'],
        invites: ['djVJJm'],
        reservedPresents: ['RDLl3K'],
    },
    {
        userName: 'Yinalin',
        email: 'wildflower@wisssh.com',
        password: 'lifeisbeautiful',
        avatar: '/avatars/Yinalin.jpg',
        wishIds: ['zLhS69'],
        wishlistIds: ['qMuBSN'],
        invites: ['djVJJm'],
        reservedPresents: [],
    },
]
