// api/mockUsers.js
export const users = [
    {
        id: '1',
        firsName: 'Htonatan',
        lastName: null,
        email: 'htonatan-the-best@wisssh.com',
        password: 'blackmamba',
        avatar: '/avatars/Htonatan.jpg',
    },
    {
        id: '2',
        firsName: 'Jari',
        lastName: 'Smith',
        email: 'jarishka@wisssh.com',
        password: 'burnmotherfucker',
        avatar: '/avatars/Jari Smith.jpg',
    },
    {
        id: '3',
        firsName: 'Jennishada',
        lastName: null,
        email: 'jennycutie@wisssh.com',
        password: 'notjenifer',
        avatar: '/avatars/Jennishada.jpg',
    },
    {
        id: '4',
        firsName: 'Manne',
        lastName: 'Kumar',
        email: 'mannekumar@wisssh.com',
        password: 'kumarkumar',
        avatar: '/avatars/Manne Kumar.jpg',
    },
    {
        id: '5',
        firsName: 'Neuroslav',
        lastName: null,
        email: 'neuroslav1488@wisssh.com',
        password: 'uroborossiaforever',
        avatar: '/avatars/Neuroslav.jpg',
    },
    {
        id: '6',
        firsName: 'Rakeishand',
        lastName: null,
        email: 'rakeishand@wisssh.com',
        password: 'iamrakeishand',
        avatar: '/avatars/Rakeishand.jpg',
    },
    {
        id: '7',
        firsName: 'Sarra',
        lastName: 'March',
        email: 'sarra16@wisssh.com',
        password: 'mylittlepony',
        avatar: '/avatars/Sarra March.jpg',
    },
    {
        id: '8',
        firsName: 'Toriahali',
        lastName: 'Yulandi',
        email: 'toyu@wisssh.com',
        password: '6nh%k$/',
        avatar: '/avatars/Toriahali Yulandi.jpg',
    },
    {
        id: '9',
        firsName: 'Yinalin',
        lastName: null,
        email: 'wildflower@wisssh.com',
        password: 'lifeisbeautiful',
        avatar: '/avatars/Yinalin.jpg',
    },
]
