// getters/index.js
import { useSelector } from 'react-redux'

export const getWishById = ( id ) => {
    const wishes = useSelector(state => state.wishes.all);
    return wishes.find( wish => wish.id === id )
}

export const getUserByUserName = ( userName ) => {
    const users = useSelector(state => state.users.all)
    return users.find( user => user.userName === userName )
}

export const getWishlistById = ( id ) => {
    const wishlists = useSelector(state => state.wishlists.all)
    return wishlists.find( wishlist => wishlist.id === id )
}

export const getDaysToEvent = ( wishlist ) => {
    if(!wishlist.date) return null;
    const eventTS = new Date( wishlist.date[2], wishlist.date[1] - 1, wishlist.date[0]).getTime();
    const nowTS = Date.now();
    return Math.ceil((eventTS - nowTS)/86400000);
}

export const sortByDateAscend = (a, b) => {
    const aTS = new Date( a.date[2], a.date[1] - 1, a.date[0]).getTime();
    const bTS = new Date( b.date[2], b.date[1] - 1, b.date[0]).getTime();
    return aTS - bTS;
}

export const sortByDateDescend = (a, b) => {
    const aTS = new Date( a.date[2], a.date[1] - 1, a.date[0]).getTime();
    const bTS = new Date( b.date[2], b.date[1] - 1, b.date[0]).getTime();
    return bTS - aTS;
}
